sentry_sdk/tracing_utils.py

```diff
@@ -739,3 +742,18 @@ def get_current_span(scope=None):
 
 if TYPE_CHECKING:
     from sentry_sdk.tracing import Span
+
+
+def finish_running_transaction(scope=None, exc_info=None, timer=None):
+    # type: (Optional[sentry_sdk.Scope], Optional[ExcInfo], Optional[Timer]) -> None
+    current_scope = scope or sentry_sdk.get_current_scope()
+    if current_scope.transaction is not None and hasattr(
+        current_scope.transaction, "_context_manager_state"
+    ):
+        if timer is not None:
+            timer.cancel()
```

Comment:
Should we maybe always cancel the timer as the first step in this function without the conditional checks on lines 749-752?

What I'm worried might happen is that if we for some reason don't cancel the timer (because there is no transaction anymore for instance), it'll still fire the `_finish_long_running_transaction` event at a later point and might potentially end up cancelling a different active transaction. IDK if that can realistically happen due to how we pass the scopes around, but probably doesn't hurt to be more cautious here.